semgrep-core/src/tainting/Dataflow_tainting.ml

```diff
@@ -349,24 +375,50 @@ let rec check_tainted_expr env exp : Taints.t * var_env =
   in
   let check_offset env = function
     | Index e -> check env e
-    | NoOffset
-    | Dot _ ->
-        (Taints.empty, env.var_env)
+    | Dot _ -> (Taints.empty, env.var_env)
   in
   let check_subexpr exp =
     match exp.e with
-    | Fetch { base = VarSpecial (This, _); offset = Dot fld; _ } ->
+    | Fetch { base = VarSpecial (This, _); offset = Dot fld :: _; _ } ->
```

Comment:
Maybe the pattern here should be `[Dot fld]` instead of `Dot  fld :: _`?

Reply:
I don't think so. When we produced lots of intermediate values, this would have worked on the IL generated by a Generic AST that had multiple offsets that _ended_ in a `Dot`. So we should replicate that.

Reply:
That's a great point!

Reply:
> That's a great point!

Wait a second, if there were multiple offsets then the `base` would not be `base = VarSpecial (This, _)` right? So I think this didn't use to work that way.